reorg of topics for new Code Security section in Security (#26887)
DataDog/documentation: content/en/security/code_security/software_composition_analysis/setup_runtime/_index.md

---
title: Software Composition Analysis Setup
disable_toc: false
aliases:
- /security/application_security/enabling/tracing_libraries/sca/
---

> ## Prerequisites

Reviewer: ```suggestion
```

> Before setting up Software Composition Analysis, ensure the following prerequisites are met:

Reviewer: ```suggestion
SCA can detect vulnerabilities that affect open source libraries running in your services based on Datadog's application telemetry. 
scan dependency management files in your repositories to statically detect open source libraries used in your codebase. 

Before setting up runtime deteciton, ensure the following prerequisites are met:
```

1. **Datadog Agent Installation:** The Datadog Agent is installed and configured for your application's operating system or container, cloud, or virtual environment.
2. **Datadog APM Configuration:** Datadog APM is configured for your application or service, and web traces (`type:web`) are being received by Datadog.

> 3. **Supported Tracing Library:** The Datadog Tracing Library used by your application or service supports Software Composition Analysis capabilities for the language of your application or service. For more details, refer to the [Library Compatibility][5] page for each ASM product.

Reviewer: ```suggestion
3. **Supported Tracing Library:** The Datadog Tracing Library used by your application or service supports Software Composition Analysis capabilities for the language of your application or service. See supported tracing libraries below:

| Capability                         | Java    | .NET     | Node.js                                          | Python        | Go              | Ruby          | PHP           |
|----------------------------------------|---------|----------|--------------------------------------------------|---------------|-----------------|---------------|---------------|
| Software Composition Analysis (SCA)    | 1.1.4   | 2.16.0   | 4.0.0                                            | 1.5.0         | 1.49.0          | 1.11.0        | 0.90.0        |

Select your application language for details about framework compatibility and feature support.

{{< partial name="security-platform/appsec-languages-sca.html" >}}
```

> ## Software Composition Analysis enablement types

Reviewer: ```suggestion
## Select a configuration option
```

> ### Quick Start Guide

Reviewer: ```suggestion
### Quick Start Guide for in-app service enablement
```

1. Navigate to the [Quick Start Guide][2]:
   1. Expand **Enable Vulnerability Detection**.
   2. Select **Open source vulnerabilities**.
   3. Select **Start Activation**.
   4. Select the services where you want to identify library vulnerabilities, and then click **Next**.
   5. Select **Enable for Selected Services**.

> ### Settings page

Reviewer: ```suggestion
### Settings page for in-app service enablement
```

Alternatively, you can enable Software Composition Analysis through the [Settings][3] page.

> 1. Navigate to the [Settings][3] page and select **Get Started** in **Software Composition Analysis (SCA)**.
> 2. For static analysis in source code, select **Select Repositories**.
> 3. Select **Add Github Account** and follow the [instructions][4] to create a new GitHub Application.
> 4. Once the GitHub account is set up, select **Select Repositories** and enable **Software Composition Analysis (SCA)**.
> 5. For runtime analysis in running services, click **Select Services**.
> 6. Choose the services where you want to identify library vulnerabilities, and select **Next**.
> 7. Select **Enable for Selected Services**.

Reviewer: ```suggestion
1. Navigate to the [Settings][3] page and select **Setup** under **Code Security** in the lefthand menu.
2. To enable runtime SCA, click the second option: **Enable runtime detection of library vulnerabilities**.
3. Choose the services where you want to identify library vulnerabilities, and select **Next**.
4. Select **Enable for Selected Services**.
```

Reviewer: Will need to do another pass in ~1 week to make sure this matches the in-app copy

> ### Datadog Tracing Libraries

Reviewer: ```suggestion
### Datadog tracing library configuration 
```

---

> title: Set up runtime SCA in your services

Reviewer: ```suggestion
title: Set up SCA in your running services
```
